### ISSUE TITLE
Generic.Formatting.MultipleStatementAlignment false positive for `match()`

### ISSUE THREAD

**Issue by LastDragon-ru** [squizlabs/PHP_CodeSniffer#3862]
## Describe the bug

`Generic.Formatting.MultipleStatementAlignment` doesn't work correctly with `match()`. 

### Code sample
```php
<?php declare(strict_types = 1);

namespace Bug;

class A {
    public function bug(bool $b): void {
        $a   = match ($b) {
            true    => 1,
            default => false,
        };
        $var = 123;

        echo $a.$var;
    }
}
```

### Custom ruleset

```xml
<?xml version="1.0"?>
<ruleset name="Bug"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="./vendor/squizlabs/php_codesniffer/phpcs.xsd">
    <file>./packages</file>
    <rule ref="Generic.Formatting.MultipleStatementAlignment"/>
</ruleset>
```

### To reproduce

1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed
```
------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
------------------------------------------------------------------------------------------
 7 | WARNING | [x] Equals sign not aligned correctly; expected 1 space but found 3 spaces
------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
------------------------------------------------------------------------------------------
```

## Expected behavior

No error.

## Versions (please complete the following information)

| | |
|-|-|
| Operating System | Win 10
| PHP version | 8.2
| PHP_CodeSniffer version | 3.7.2
| Standard | custom
| Install type | Composer (global/local)

## Please confirm:

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [ ] I have verified the issue still exists in the `master` branch of PHP_CodeSniffer.

**Comment by jrfnl** [comment]
> In the example $a and $var don't belong to the same "block of related assignments", or at least is my assumption

That's actually not true. They do belong to the same block of assignments, multi-line argument or not.
A block of assignments is only broken by a blank line a comment or another type of non-assignment statement.

While opinions may vary whether that is correct, the sniff should at least be consistent and for consistency, the above should be seen as a bug.

---
<details>
  <summary>Expanded code sample showing other multiline assignments</summary>

```php
<?php declare(strict_types = 1);

namespace Bug;

class A {
    public function bug(bool $b): void {
        $a   = match ($b) {
            true    => 1,
            default => false,
        };
        $var = 123;
        $a   = match ($b) {
            true    => 1,
            default => false,
        };

        $a   = function ($b) {
            return $b;
        };
        $var = 123;
        $a   = function ($b) {
            return $b;
        };

        $a   = 'multi-'
            . 'line';
        $var = 123;
        $a   = 'multi-'
            . 'line';

        echo $a.$var;
    }
}
```

Result:
```
----------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
----------------------------------------------------------------------------------------------
 7 | WARNING | [x] Equals sign not aligned correctly; expected 1 space but found 3 spaces
   |         |     (Generic.Formatting.MultipleStatementAlignment.IncorrectWarning)
----------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------------------------------
```

In other words: the first match on line 7 is treated inconsistently compares to other multi-line assignments, which all expect alignment.

</details>

---
